FirebaseAuth/Sources/Swift/AuthProvider/AuthProviderID.swift

```diff
@@ -0,0 +1,57 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import Foundation
+
+/// Enumeration of the available Auth Provider IDs.
+public struct AuthProviderID: Equatable {
+  public let rawValue: String
```

Comment:
Last thing, after locking down the initialization by removing `RawRepresentable`, I'm sort of going back and forth now with whether the `rawValue` name is misleading since you would think the type would be `RawRepresentable` with a property of that name.

WDYT? Do you prefer `identifier` or `id` over `rawValue`?

Reply:
What about plain `value`?

Reply:
Yes, `value` SGTM.

Reply:
I think it's unlikely this struct needs to conform to `RawRepresentable` in the future, but one pro to how it is currently is that a `rawValue` property name makes it very easy to add `RawRepresentable` conformance without needing to deprecate a separate variable name (e.g. `value`).

Reply:
Sorry to muddy the water, I could swing both ways really. `rawValue` is maybe slightly less idiomatic but more future proof. I'll defer to your preference!

Reply:
OK, `rawValue` might be more aligned with expectations - so I'll leave it. Thanks.